Convert specific file-specific extern functions to static scope.

MasaoFujii opened the pull request:
Previously, there were several functions declared as extern in postgres_fdw_plus.h that were only used in connection.c or postgres_fdw_plus.c. To enhance clarity and encapsulation, this commit converts these functions to static scope within their respective files. As a result, the extern declarations for these functions are removed from postgres_fdw_plus.h, further clarifying their limited visibility and usage.

YKatsuragi commented:
Thank you for the PR!

Let me ask a question.

> +/* postgres_fdw_plus.c */
>  extern void pgfdw_arrange_read_committed(bool xact_got_connection);
>  extern bool pgfdw_xact_two_phase(XactEvent event);

Does this `/* postgres_fdw_plus.c */` mean the file where these functions are defined?
If so, I think it is better for this PR to place `DefineCustomVariablesForPgFdwPlus(void);` and `pgfdw_abort_cleanup(ConnCacheEntry *entry, bool toplevel);` in the section of postgres_fdw_plus.c.

MasaoFujii commented:
@YKatsuragi 
> Does this /* postgres_fdw_plus.c */ mean the file where these functions are defined?

Yes.

> If so, I think it is better for this PR to place DefineCustomVariablesForPgFdwPlus(void); and 
> pgfdw_abort_cleanup(ConnCacheEntry *entry, bool toplevel); in the section of postgres_fdw_plus.c.

Agreed. I have updated the patch accordingly.

I noticed that there are some global variables in postgres_fdw_plus.c that are only used within that file but are declared as extern variables in postgres_fdw_plus.h. I have changed their scope to static.

Additionally, I found that the macro PreparedXactCommand() is only used in postgres_fdw_plus.c, so I have moved it from postgres_fdw_plus.h to the .c file.

YKatsuragi commented:
Thank you for updating the PR!

Looks good to me.
